src/collection.js

```diff
@@ -322,7 +323,51 @@ export default class Collection {
   listRecords(options={}) {
     const path = endpoint("record", this.bucket.name, this.name);
     const reqOptions = this._collOptions(options);
-    return this.client.paginatedList(path, options, reqOptions);
+    if (options.hasOwnProperty("at")) {
+      return this._getSnapshot(options.at);
+    } else {
+      return this.client.paginatedList(path, options, reqOptions);
+    }
+  }
+
+  /**
+   * @private
+   */
+  _getSnapshot(at) {
+    if (!Number.isInteger(at) || at <= 0) {
+      throw new Error("Invalid argument, expected a positive integer.");
+    }
+    // TODO: we process history changes forward, while it would probably be more
+    // efficient and accurate to process them backward.
+    return this.bucket.listHistory({
+      pages: Infinity, // all pages up to target timestamp are required
+      sort: "-target.data.last_modified",
+      filters: {
+        resource_name: "record",
+        collection_id: this.name,
+        "max_target.data.last_modified": String(at),
+      }
+    })
+      .then(({data: changes}) => {
+        const seenIds = new Set();
+        let snapshot = [];
+        for (const {target: {data: record}} of changes) {
+          if (record.deleted) {
+            seenIds.add(record.id); // ensure not reprocessing deleted entries
+            snapshot = snapshot.filter(r => r.id !== record.id);
+          } else if (!seenIds.has(record.id)) {
+            seenIds.add(record.id);
+            snapshot.push(record);
+          }
+        }
+        return {
+          last_modified: Math.max.apply(null, snapshot.map(r => r.last_modified)),
```

Comment:
Done in https://github.com/Kinto/kinto-http.js/pull/161/commits/ce8a1c763664906ce85afb91f37e912a0488a061